src/wallet/walletdb.cpp

```diff
@@ -546,96 +546,17 @@ ReadKeyValue(CWallet* pwallet, DataStream& ssKey, CDataStream& ssValue,
             }
         } else if (strType == DBKeys::WATCHS) {
             wss.nWatchKeys++;
-            CScript script;
-            ssKey >> script;
-            uint8_t fYes;
-            ssValue >> fYes;
-            if (fYes == '1') {
-                pwallet->GetOrCreateLegacyScriptPubKeyMan()->LoadWatchOnly(script);
-            }
         } else if (strType == DBKeys::KEY) {
-            if (!LoadKey(pwallet, ssKey, ssValue, strErr)) return false;
             wss.nKeys++;
         } else if (strType == DBKeys::MASTER_KEY) {
             if (!LoadEncryptionKey(pwallet, ssKey, ssValue, strErr)) return false;
         } else if (strType == DBKeys::CRYPTED_KEY) {
-            if (!LoadCryptedKey(pwallet, ssKey, ssValue, strErr)) return false;
```

Comment:
In commit "walletdb: Refactor legacy wallet record loading into its own function" (8648511b567dfcdea7ffa5ac4595a43a768f5525)

It looks like in this commit, the side-effect in previous commit "walletdb: Refactor key reading and loading to its own function" (e048fe71038caddb49a5875727a14e05f6edb0d3) of changing the `wss.nKeys` value in case of key errors is partially reverted, because now the number of keys is incremented unconditionally, while the previous commit added more conditions. 

Would suggest changing previous commit to just increment `wss.nKeys++` unconditionally, so that behavior doesn't change here and this commit is simpler.

Same comment applies to `wss.nCKeys` and commit "walletdb: Refactor crypted key loading to its own function" (4940c1cb3dca97faa2b6a4ddfab889cd4b6f00f4)

The `wss.nKeyMeta` count also changes here if there is a deserialization exception. It would be good to note this in the commit message for this commit.